test/jdk/tools/jar/JarEntryTime.java

```diff
@@ -123,6 +124,16 @@ public static void realMain(String[] args) throws Throwable {
                            "cf", jarFile.getName(), dirOuter.getName()) == 0);
         check(jarFile.exists());
 
+        // Make a jar file from that directory structure with
+        // --source-date set to epoch seconds 1647302400 (15/03/2022)
+        long sourceDate = 1647302400L;
```

Comment:
@LanceAndersen can you clarify your meaning here please? Are you referring to negative sourceDate? I put in the CSR that it was seconds "since" epoch akin to SOURCE_DATE_EPOCH, which as far as I can tell does not support negative values, as some time implementations won't support that. 
But this does imply I should add some -ve validation checks.

Reply:
I think we could support -ve values, though i'd need to alter a couple of things

Reply:
@LanceAndersen java File times can't be before the epoch, but having a test before dostime 1980 would be useful

Reply:
> 

The change to sun/tools/jar/GNUStyleOptions.java does not prevent a negative value which can be set via ZipEntry similar to:
```
    public void testOfEpochSecond() {
        var ze = new ZipEntry("test");
        for(var i =  0; i < 100; i++) {
            var time = LocalDateTime.ofEpochSecond(-i, 0, ZoneOffset.UTC);
            ze.setTimeLocal(time);
            System.out.printf(
                    "time= %s, Zip Entry time= %s%n", time, ze.getTimeLocal());
        }
    }
```
If the intent is to not support dates prior to the Epoch then GNUStyleOptions should throw an Exception in this case.

Reply:
Added more tests, and negative check